### ISSUE TITLE
Display latest commits on FT website homepage

### ISSUE THREAD

**Issue by Spaak** [fieldtrip/website#8]

**Comment by robertoostenveld** [comment]
from https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-log.html I got this 

  git log --pretty=format:"- commit: %h%n  name: %cN%n  subject: %s%n  date: %ci%n  hash: %H"

which results in a list like this (here for the website, but for fieldtrip itself it would be similar)

```
- commit: 960a44e
  name: Robert Oostenveld
  subject: include google analytics tracking, updated README
  date: 2018-11-20 14:44:30 +0100
  hash: 960a44eb4e8a2516873417ea2a8aaf7ee5ecc2c4
- commit: fd624a5
  name: Robert Oostenveld
  subject: updated dependencies with "gem install bundler" to update bundler, followed by "bundle install"
  date: 2018-11-20 14:21:23 +0100
  hash: fd624a57913de35bf5de3a0588c098941514cc6a
- commit: 719a31b
  name: Robert Oostenveld
  subject: added images to the asset folder, updated publish_website script
  date: 2018-11-20 14:16:42 +0100
  hash: 719a31bc1c1cb8407fd7d0363ec5a25190e6a2fc
```

We could write this to a yaml file in _data, and auto-generate a nicely formatted list, just like the tags and navigation includes.  

To have this run automatically, we could add it to the https://github.com/fieldtrip/webhook/blob/master/github.sh which runs upon every commit.

**Comment by Spaak** [comment]
I'm wondering whether it might not make more sense to get this kind of information client-side using JavaScript. I can see several advantages to that approach:
- Avoids bloating the post-commit hooks
- Information is always up to date; no regeneration delay between the commit on GitHub and it displaying on the website. This adds to the feeling of a 'live' project.

**Comment by robertoostenveld** [comment]
Rather than each commit, we might want to announce the latest release (and the commits that it contains). The release process is now automatic, and only when all commits on master pass the full dashboard batch, the master is merged in release and given a YYYYMMDD tag. This is (partially) explained on https://github.com/fieldtrip/fieldtrip/issues/1026

**Comment by robertoostenveld** [comment]
Releases could be announced as news items, i.e. as a Jekyll post with the category `news`. See  https://github.com/fieldtrip/website/tree/master/_posts for examples and https://raw.githubusercontent.com/fieldtrip/website/master/index.md for how those posts are included in the news overview.

**Comment by robertoostenveld** [comment]
Perhaps this is something to discuss with @elshafeh, who wants to change the automatic tweets that are sent. It would be nice if they were in line, e.g. to showcase every new release (not every new commit).

**Comment by elshafeh** [comment]
hi everyone , sorry i missed today's meeting. more than happy to discuss this on the next zoom! stay safe.

**Comment by Spaak** [comment]
I agree, it would be great not to tweet every commit. Should I look into making this happen?

**Comment by robertoostenveld** [comment]
Yes please! 

@elshafeh has the twitter details, which are stored in a `secret.sh` file like https://github.com/fieldtrip/automation/blob/master/secret-example.sh

The tweets are now sent from https://github.com/fieldtrip/automation/blob/master/webhook.js which is a small nodes server running on a Raspberry Pi in my basement.

It would make sense to make the automatic message following a release part of https://github.com/fieldtrip/automation/blob/master/release-ftp.sh. That script runs every evening from a cron job at 23:45, which is (in general) after the daily dashboard  run has completed running. The release only happens if all test scripts pass.

**Comment by robertoostenveld** [comment]
please do make sure that the twitter API secret details are not spilled on the internet in a pull request or so. That recently happened, and I had to reset all keys.

**Comment by elshafeh** [comment]
Here's where we are https://github.com/fieldtrip/automation/pull/7

**Comment by robertoostenveld** [comment]
I have made an alternative implementation, using a news item that can be displayed on the front page just like all other news items. The Bash script that creates the news item is `release-newsitem.sh` in https://github.com/fieldtrip/automation

It makes a branch of the website, creates the news item, commits, pushes the branch and (although disabled at the moment) creates a pull request. At that moment it is up to the website repository maintainers to make manual improvements to the message (e.g. write a more human-readable summary rather than the list of commits) or accept the news item as it is. The PR can be merged via the GitHub web interface and then it will appear on the FieldTrip website.

I have created a number of these news items of recent releases to get a feel for how the automated version (with a list of commits) would look like. See https://github.com/fieldtrip/website/branches
